src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/StartupObjectsEnumProvider.cs

```diff
@@ -94,6 +94,9 @@ public async Task<ICollection<IEnumValue>> GetListedValuesAsync()
                 }));
             }
 
+            // Remove My.MyApplication entry if any.
+            enumValues = enumValues.Where(ep => !ep.Name.Contains("My.MyApplication")).ToList();
```

Comment:
_Technically_ this code runs for C# too. It seems unlikely that would ever be an issue.

This can also be slightly more efficient if you use [`List.RemoveAll`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.removeall?view=net-7.0):

```suggestion
            enumValues.RemoveAll(ep => !ep.Name.Contains("My.MyApplication"));
```

It's more efficient because it doesn't allocate a new list, and if nothing matches the predicate then no memory is modified at all, whereas the previous code would always create a full copy.

Reply:
> Technically this code runs for C# too. It seems unlikely that would ever be an issue.

That's what I thought too, this value only makes sense in VB projects. These entries come from `IEntryPointFinderService` from Roslyn; it might be the case that we need to make the change there instead of here, but I'm not sure.

Reply:
It's definitely worth investigating whether the filtering should live on the Roslyn side. Ideally we don't get involved in these kinds of language level policies.